### PR TITLE
Downstream guide

### LINKED ISSUE #665: Packit Onboarding Guide TOC
I were looking for table of contents in the onboarding guide. Who does it expect on the right side? I certainly did not, because I kind of expect advertisements there. It seems to me there should be ToC at least for GitHub, Gitlab difference. With images this page seems too long. It contains too many header levels. Github has 3 subsections in section 1. at the same level as GitLab.

I would recommend splitting onboarding guide into more separate pages. ToC should be in the center below the main title, maybe with button to expand also subsections. The page should have visually different h2, h3 and h4. They are too similar IMO, fast scrolling does not hint where exactly github ends.

I searched for information how to integrate with dist-git only, because I am not upstream for most of my packages and have no direct commit rights. Unfortunately I haven't found any trace of such feature, which I heard is supported from Mirek. 

I have found some mention at:
https://packit.dev/docs/fedora-releases-guide/#pull-from-upstream-job

But I would expect it more straightforward. Maybe onboarding guide should mention early what kind of access do I need in upstream repository used. It seems to me it still requires admin access to repository and does not mention it explicitly before it starts describing steps, which I cannot do without it. If it can do something useful without that, I think it should say so before describing steps required on each forge.

### LINKED ISSUE #802: Write a blog post about downstream jobs targeting Fedora package maintainers
We do have [Fedora releases guide](https://packit.dev/docs/fedora-releases-guide) but Fedora maintainers are still getting confused by seeing upstream jobs mentioned. Write a blog post that:

- [ ] demonstrates a basic configuration (`pull_from_upstream`, `koji_build`, `bodhi_update`) in dist git
- [ ] explains how to verify the config with CLI, how to test/run locally
- [ ] explains how/when the jobs are triggered by the service, how to trigger them manually
- [ ] explains `issue_repository`
- [ ] does not mention `propose_downstream` or any other upstream jobs

As a bonus:
- [ ] explains why `upstream_project_url` is needed (and `upstream_tag_template`)

### LINKED ISSUE #825: Automate CLI help generation
This is about automating the generation of help messages that are shown in https://packit.dev/docs/cli, e.g. https://packit.dev/docs/cli/init#help.

Currently, we copy-paste it and it gets outdated very easily.

### PR DESCRIPTION
**Pull request by lbarcziova**
Fixes #802 
Fixes #825 
Fixes #665 

When reviewing, it is probably best to go commit by commit.

TODO:

- [ ] agree on the restructure
- [x] fix broken links (check also other repos)
- [x] create docs for CLI pull-from-upstream 

RELEASE NOTES BEGIN

N/A

RELEASE NOTES END

### CODE REVIEW

File: docs/cli/pull-from-upstream.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+title: pull-from-upstream
+date: 2024-02-22
+sidebar_position: 8
+---
+
+# `packit pull-from-upstream`
+
+This is a detailed documentation for the update functionality of packit. The
+command creates a new pull request in Fedora (or CentOS Stream) using a selected or latest
```
Comment (majamassarini) [inline docs/cli/pull-from-upstream.md:10]:
```suggestion
command creates a new pull request in Fedora (or CentOS Stream) using the selected or the latest
```
Not sure if this is better but "a selected" does not sounds to me...

File: docs/cli/pull-from-upstream.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+upstream release. This command is meant to be called from **dist-git repository**.
+
+
+## Requirements
+
+* Upstream git repository on GitHub/GitLab.
+* Upstream release (read, git tag) where version in spec file is equivalent to
```
Comment (majamassarini) [inline docs/cli/pull-from-upstream.md:17]:
```suggestion
* Upstream release (both release and git tag) where version in spec file is equivalent to
```

Not sure if this is what you mean.

**Review by majamassarini — COMMENTED** [review]
Really nice the script for updating the docs. Thanks!

File: .github/workflows/regenerate-cli-help-messages.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+name: Regenerate help messages of CLI commands
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * *"
```
[the remark this answers is not in the document]
Comment (majamassarini) [reply .github/workflows/regenerate-cli-help-messages.yml:6]: Yes, `pip install packitos` (right?) should be more aligned with what the users have.

File: docs/guide.md [hunk 4]
```diff
@@ -437,37 +463,71 @@ or the shorter version
...
+
+[VM Image builds](/docs/configuration/upstream/vm_image_build) are not triggered automatically at all.
+To trigger them, you need to post a pull-request comment:
+
+    /packit vm-image-build
+
+every time.
+
+
+:::caution
```
Comment (TomasTomecek) [inline docs/guide.md:507]: I really like this change that most of the retriggering commands are here now.

**Review by TomasTomecek — APPROVED** [review]
looks like a great improvement, only reviewed it briefly

File: docs/cli/pull-from-upstream.md (hunk 2 above)
Comment (lbarcziova) [reply docs/cli/pull-from-upstream.md:17]: I just mostly copied what we already have for `propose-downstream`, but in fact git tag should be enough from CLI

**Pull request merged**